### DISCUSSION TITLE
Error while running the Qualification tool

### DISCUSSION THREAD

**Discussion by SoumyaB57** [NVIDIA/spark-rapids#5447] [General] [answered]
I wanted to run the qualification tool with the spark event log stored on the local file system.
I have downloaded the Jar file mentioned [here](https://nvidia.github.io/spark-rapids/docs/spark-qualification-tool.html#step-2-run-the-qualification-tool)
But I am getting the following error
`Error: Could not find or load main class com.nvidia.spark.rapids.tool.qualification.QualificationMain
`

**Comment by tgravescs** [comment]
It sounds like your jar is not in the class path.  What command did you use and what is your jar named?

You should run something like this, but note to replace <version> with the actual version of the jar you downloaded.  Also make sure $SPARK_HOME is set to your installation of Spark.

```
 java -cp rapids-4-spark-tools_2.12-<version>.jar:$SPARK_HOME/jars/*
        com.nvidia.spark.rapids.tool.qualification.QualificationMain [options]
        <eventlogs | eventlog directories ...>
```

**Reply by SoumyaB57** [reply]
I have installed spark 3.0.3 previously and SPARK_HOME is defined as `/Users/{MyName}/Documents/Dev/Apache-Spark/spark-3.0.3-bin-hadoop2.7` and I have downloaded `rapids-4-spark-tools_2.12-22.04.0-javadoc.jar`
So my command is `java -cp rapids-4-spark-tools_2.12-22.04.0-javadoc.jar:$SPARK_HOME/jars/* com.nvidia.spark.rapids.tool.qualification.QualificationMain eventlogs `
But currently, the error I am getting is 
`zsh: no matches found: rapids-4-spark-tools_2.12-22.04.0-javadoc.jar:/Users/{MyName}/Documents/Dev/Apache-Spark/spark-3.0.3-bin-hadoop2.7/jars/*`

**Comment by jlowe** [comment, marked as answer]
You will need to escape the wildcard from trying to be expanded by the shell.  We want the JVM to see the wildcard, not the shell, and zsh is being too eager about trying to expand it and failing if it cannot.  For example, you can quote it like this:
```
java -cp "rapids-4-spark-tools_2.12-22.04.0-javadoc.jar:$SPARK_HOME/jars/*" com.nvidia.spark.rapids.tool.qualification.QualificationMain eventlogs
```

**Reply by SoumyaB57** [reply]
Now it is showing the error I mentioned previously
 `Error: Could not find or load main class com.nvidia.spark.rapids.tool.qualification.QualificationMain
Caused by: java.lang.ClassNotFoundException: com.nvidia.spark.rapids.tool.qualification.QualificationMain`
My command `java -cp "rapids-4-spark-tools_2.12-22.04.0-javadoc.jar:$SPARK_HOME/jars/*" com.nvidia.spark.rapids.tool.qualification.QualificationMain eventlogs      `

**Reply by tgravescs** [reply]
you are using the java doc.jar (rapids-4-spark-tools_2.12-22.04.0-javadoc.jar), you need the regular tools jar. [rapids-4-spark-tools_2.12-22.04.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark-tools_2.12/22.04.0/rapids-4-spark-tools_2.12-22.04.0.jar)

**Reply by SoumyaB57** [reply]
Thank you @tgravescs and @jlowe for the reply, this issue is solved now.
